clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -779,7 +787,10 @@ protected void close(Timer timer) {
             // Synchronize after closing the heartbeat thread since heartbeat thread
             // needs this lock to complete and terminate after close flag is set.
             synchronized (this) {
-                if (leaveGroupOnClose) {
+                // Starting from 2.2, only dynamic members will send LeaveGroupRequest to the broker,
+                // consumer with valid group.instance.id is viewed as static member that never sends LeaveGroup,
+                // and the membership expiration is only controlled by session timeout.
+                if (this.groupInstanceId.equals(JoinGroupRequest.UNKNOWN_GROUP_INSTANCE_ID)) {
```

Comment:
Should we add a test which exercises this code path? I think it's critical for the KIP that this works

Reply:
Sounds good!